From a3ec578ff5afb88dddec526bb21981343f7429ea Mon Sep 17 00:00:00 2001
From: Sebastian Lackner <sebastian@fds-team.de>
Date: Sun, 6 Mar 2016 22:08:41 +0100
Subject: Revert "wined3d: Avoid surface_from_resource() in
 swapchain_gl_frontbuffer_updated()."

This reverts commit c059191e0fe795e4643e56bb2ab806c9ce994055.
---
 dlls/wined3d/swapchain.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/wined3d/swapchain.c b/dlls/wined3d/swapchain.c
index 299a47a..9b8d04c 100644
--- a/dlls/wined3d/swapchain.c
+++ b/dlls/wined3d/swapchain.c
@@ -652,7 +652,7 @@ static void swapchain_gl_frontbuffer_updated(struct wined3d_swapchain *swapchain
     struct wined3d_surface *surface;
     struct wined3d_context *context;
 
-    surface = swapchain->front_buffer->sub_resources[0].u.surface;
+    surface = surface_from_resource(swapchain->front_buffer->sub_resources[0].resource);
     context = context_acquire(swapchain->device, surface);
     surface_load_location(surface, context, surface->container->resource.draw_binding);
     context_release(context);
-- 
2.7.1

